### DISCUSSION TITLE
Vector Host Statisitics

### DISCUSSION THREAD

**Discussion by tastyfrankfurt** [vectordotdev/vector#8309] [General]
Hey Team,

We want to be able to monitor disk usage percentage for our systems on OpenSearch, wondering if the best path is feature request into the host metrics to calculate this, or if there is another way, as having trouble aggregating 2 events together in OpenSearch.

Thanks
Tasty

**Comment by spencergilbert** [comment]
The `host_metrics` source should provide free and total bytes for filesystems. What attic metrics are you looking for?

Ref: https://vector.dev/docs/reference/configuration/sources/host_metrics/#filesystem_free_bytes

**Reply by tastyfrankfurt** [reply]
Hey @spencergilbert, it does but the issue we are having is we use opensearch and because each gauge comes in separate events we cannot correlate them to how full the disk is as we need to link 2 events. Looking at metric beat it has the fields in 1 event and they also have percentage full.  See links below,
system.filesystem.used.pct - https://www.elastic.co/guide/en/beats/metricbeat/current/exported-fields-system.html

Unless you know of a way i can aggregate these events together then we cannot measure how full a partiction is as we cannot link used/free to total on this platform.

**Reply by spencergilbert** [reply]
Gotcha - that makes sense! It looks like the library we use also exposes a [`ratio` calculation](https://docs.rs/heim/0.0.11/heim/disk/struct.Usage.html#method.ratio) that we don't use today, that may be suitable for what you're looking for?

If it is, I don't think it should be very difficult to include that as an exposed metric as well.

**Reply by tastyfrankfurt** [reply]
@spencergilbert Yes that certainly looks like the one and looks pretty straight forward looking at the code base to, if anything more editing to be done adding it to the doco than to the code, then again it usually is with features on well written code.  I guess next questino is lead times, on this, assuming if wont make 0.15 but should be able to get into 0.16 in say 3 months?

**Reply by spencergilbert** [reply]
_Yeah_ probably not the 0.16 release which is mid release now 😉 - but our next release is planned for August 24, and of course depending on your position/policies we release nightly images/binaries that could be used as soon as the code is merged.

I'll open an issue to add the `ratio` value to the metrics exposed by `host_metrics`.

**Reply by tastyfrankfurt** [reply]
hehe we tried to run a nightly release but we hit a few bugs :-) so the manager is a bit nervous about it. Thanks for your help @spencergilbert again 👍 , ill let them know to expect after august 24.

**Reply by tastyfrankfurt** [reply]
One more thing whilst i have your attention which release is the Azure Storage Account sink due in? We have another use case that we tested with the nightly build and had issues with, the azure sink worked well though.

**Reply by spencergilbert** [reply]
🙂 that'll be in the 0.16 release which should be pushed out this morning/today.

**Reply by tastyfrankfurt** [reply]
Awesome :-)
